sycl/source/detail/graph_impl.cpp

```diff
@@ -285,17 +268,25 @@ graph_impl::add(sycl::detail::CG::CGTYPE CGType,
       MemObj->markBeingUsedInGraph();
     }
     // Look through the graph for nodes which share this requirement
-    for (auto &NodePtr : MRoots) {
-      checkForRequirement(Req, NodePtr, UniqueDeps);
+    for (auto Node : MNodeStorage) {
```

Comment:
`auto &Node` normally capture by reference in these loops unless there's a reason to do otherwise